docs/support/troubleshooting.md

```diff
@@ -49,60 +48,98 @@ To resolve this, try either one of the following methods:
 
 !!! warning
 
-    Note that some components in `autoware.universe` don't work without CUDA.
+    Note that some components in Autoware Universe require CUDA.
 
     Also, although Autoware could work with different versions of CUDA libraries as well,
     remember it is not officially supported and may sometimes be broken.
 
-## How to resolve build errors?
+## Build errors
 
-When you get build errors, follow the instruction below depending on the version you use.
+### Insufficient memory
 
-### In case you use the latest version
+Building Autoware requires a lot of memory, and your machine can freeze or crash if memory runs out during a build. To avoid this problem, 16-32GB of swap should be configured. For details on how to configure a swap file, refer to [this Digital Ocean guide](https://www.digitalocean.com/community/tutorials/how-to-add-swap-space-on-ubuntu-20-04).
```

Comment:
I think it's better to write the commands here instead of the link. @LalithVipulananthan What do you think?

```suggestion
Building Autoware requires a lot of memory, and your machine can freeze or crash if memory runs out during a build. To avoid this problem, 16-32GB of swap should be configured.

```bash
# Optional: Check the current swapfile
free -h

# Remove the current swapfile
sudo swapoff /swapfile
sudo rm /swapfile

# Create a new swapfile
sudo fallocate -l 32G /swapfile
sudo chmod 600 /swapfile
sudo mkswap /swapfile
sudo swapon /swapfile

# Optional: Check if the change is reflected
free -h
```
```

Reply:
You can use some different commands, but I personally use the commands above.

Reply:
Also, it's okay to keep the link as a reference.